synapse/media/oembed.py

```diff
@@ -109,18 +109,20 @@ def autodiscover_from_html(self, tree: "etree.Element") -> Optional[str]:
             The URL to use for oEmbed information, or None if no URL was found.
         """
         # Search for link elements with the proper rel and type attributes.
-        for tag in tree.xpath(
-            "//link[@rel='alternate'][@type='application/json+oembed']"
+        for tag in cast(
+            List["etree._Element"],
+            tree.xpath("//link[@rel='alternate'][@type='application/json+oembed']"),
         ):
             if "href" in tag.attrib:
-                return tag.attrib["href"]
+                return cast(str, tag.attrib["href"])
```

Comment:
`attrib` is a [special `_Attrib` class](https://github.com/lxml/lxml-stubs/blob/f0290b545e2c694c66934ec36b6e77b990cceb1e/lxml-stubs/etree.pyi#L301-L331), the `__getitem__` takes an `_AnyStr` and returns an `_AnyStr`, I think this works if you give a string you get a string (and if you give bytes, you get bytes). Anyway, we always get a `str` back but mypy thinks it could also be `bytes`.